### PR TITLE
WT-9867 Only run.py should be executable

### PR DESCRIPTION
**Pull request by adelinexchen**
All files in test/suite should be non executable excluding run.py
Additionally, created test called s_executable the checks only run.py is an executable

### CODE REVIEW

File: dist/s_executable [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+# cd into the directory u want
```
Comment (mariammojid) [inline dist/s_executable:2]: np you*

Comment (adelinexchen) [reply dist/s_executable]: oops forgot to remove my comments, fixed https://github.com/wiredtiger/wiredtiger/pull/10063/commits/11f569d291f2fe26dd6ddf8fe76a81a7150e02a6

**Comment by adelinexchen** [comment]
Unsure of what to put in my echo message for when the test passes / fails

File: dist/s_executable [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+cd ../test/suite
+if [[ $(find . -executable -type f | grep -v -w "./run.py") ]]; then
+    echo "failed test"
```
Comment (korteland) [inline dist/s_executable:4]: My suggestion here would be to echo nothing at all in the success case, and to print the filename (and problem) when failing. e.g. `echo "${file} incorrectly marked executable"`

File: dist/s_executable [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
Comment (korteland) [inline dist/s_executable:1]: Nitpick: `#!/usr/bin/env bash` (This calls `env`, which will locate `bash` itself - there might not be a `bash` in `/bin`, or they might be using a different, bash-compatible shell.)

File: test/suite/test_alter01.py [hunk 4]
```diff
@@ -176,5 +176,4 @@ def test_alter01_access(self):
                         self.verify_metadata(access_str)
                         self.verify_metadata(cache_str)
 
-if __name__ == '__main__':
-    wttest.run()
+
```
Comment (korteland) [inline test/suite/test_alter01.py:179]: As discussed IRL - would be better to avoid a second blank line here.

Comment (adelinexchen) [reply test/suite/test_alter01.py]: fixed https://github.com/wiredtiger/wiredtiger/pull/10063/commits/93b3582cb7e017ca29b1bf13af7053566d4db825

File: dist/s_executable (hunk 3 above)
Comment (adelinexchen) [reply dist/s_executable]: fixed https://github.com/wiredtiger/wiredtiger/pull/10063/commits/4bf8930851b6ae2db75e2371cea38203d705710c

File: test/suite/test_compat04.py [hunk 5]
```diff
@@ -120,9 +120,3 @@ def test_compat04(self):
         config_str = 'compatibility=(release=%s,require_max=%s)' % (self.rel, self.rel)
         conn = self.wiredtiger_open('.', config_str)
         conn.close()
-
-if __name__ == '__main__':
-    wttest.run()
-
-if __name__ == '__main__':
-    wttest.run()
```
Comment (korteland) [inline test/suite/test_compat04.py:128]: This is funny - well-spotted!

File: dist/s_executable
Comment (korteland) [inline dist/s_executable:1]: Where does this get called from? Does `s_all` automatically know to run scripts that start with `s_`?

File: test/suite/test_backup19.py [hunk 6]
```diff
@@ -106,5 +106,4 @@ def test_backup19(self):
         self.take_full_backup(self.home_full)
         self.take_incr_backup(self.home_incr)
         self.compare_backups(self.uri, self.home_full, self.home_incr, str(self.bkup_id))
-if __name__ == '__main__':
-    wttest.run()
+
```
Comment (korteland) [inline test/suite/test_backup19.py:109]: This one still needs a look - will leave a comment on the other couple of random files that need some attention.

File: test/suite/test_cursor10.py [hunk 7]
```diff
@@ -123,5 +123,4 @@ def test_index_projection(self):
             count += 1
         self.assertEqual(count, self.nentries)
         cursor.close()
-if __name__ == '__main__':
-    wttest.run()
+
```
Comment (korteland) [inline test/suite/test_cursor10.py:126]: Also needs checking

File: test/suite/test_schema03.py [hunk 8]
```diff
@@ -568,5 +568,4 @@ def check_entries(self, tc):
                 self.check_one(full_idxname, cursor, key, val)
             cursor.close()
             self.show_python("cursor.close()")
-if __name__ == '__main__':
-    wttest.run()
+
```
Comment (korteland) [inline test/suite/test_schema03.py:571]: Also needs checking. I think the pattern here is cases where the "if main" isn't separated from the earlier code by a line break.

File: test/suite/test_timestamp26.py [hunk 9]
```diff
@@ -543,5 +543,4 @@ def test_in_memory_ts(self):
 
         self.ignoreStdoutPatternIfExists('/unexpected timestamp usage/')
         self.ignoreStdoutPatternIfExists('/no timestamp provided/')
-if __name__ == '__main__':
-    wttest.run()
+
```
Comment (korteland) [inline test/suite/test_timestamp26.py:546]: Also needs checking

Comment (adelinexchen) [reply test/suite/test_timestamp26.py]: fixed https://github.com/wiredtiger/wiredtiger/pull/10063/commits/664998bfc683a9df419b0c95ec375f3db616e36e

File: dist/s_executable
Comment (adelinexchen) [reply dist/s_executable:1]: fixed maybe https://github.com/wiredtiger/wiredtiger/pull/10063/commits/05764e6c75eeb73ef8b6bd314ee66281461434cd

**Review by korteland — APPROVED** [review]
LGTM! Nice work :-)

**Review by marcbutler — APPROVED** [review]
I won't pretend I reviewed this in depth 515 files!!! :) But I did checkout and test and it all worked as expected. LGTM

**Review by Sean04 — APPROVED** [review]
LGTM!

**Comment by adelinexchen** [comment]
evergreen merge

**Pull request merged**